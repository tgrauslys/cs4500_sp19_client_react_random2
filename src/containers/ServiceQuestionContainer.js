import React from 'react'
import ServiceQuestions from '../components/ServiceQuestions'
class ServiceQuestionContainer extends React.Component {
    constructor(props) {
        super(props)
        this.serviceQuestionService = this.props.service
        this.currentPage = this.props.currentPage
        this.itemCount = this.props.itemCount
        this.optionValues = this.props.optionValues
        this.state = {
            serviceQuestions: [],
            optionValues: [],
            currentPage: 0,
            itemCount: 0,
            first: true,
            last: false
        }
    }
    setPage = (e, pageNumber) => {
        const itemCount = (e && e.target && e.target.value) ? e.target.value : this.state.itemCount
        const newPageNumber = (typeof pageNumber === "number") ? pageNumber : 0
		this.serviceQuestionService.findQuestionPage(newPageNumber, itemCount)
            .then(serviceQuestions => {
                this.setState({
                    serviceQuestions: serviceQuestions.content,
					currentPage: newPageNumber,
                    itemCount: itemCount,
                    first: serviceQuestions.first,
                    last: serviceQuestions.last
                })})
    }
    componentDidMount() {
        this.serviceQuestionService
            .findQuestionPage(this.currentPage, this.itemCount)
            .then(serviceQuestions =>
                this.setState({
                    serviceQuestions: serviceQuestions.content,
                    optionValues: this.optionValues,
                    currentPage: serviceQuestions.pageable && serviceQuestions.pageable.pageNumber,
                    itemCount: serviceQuestions.size,
                    first: serviceQuestions.first,
                    last: serviceQuestions.last
                })
            )
    }
    render() {
        return (
            <div>
                <ServiceQuestions
                    serviceQuestions={this.state.serviceQuestions}
                    optionValues={this.state.optionValues}
                    currentPage={this.state.currentPage}
                    itemCount={this.state.itemCount}
                    first={this.state.first}
                    last={this.state.last}
                    setPage={this.setPage}/>
            </div>
        )
    }
}

export default ServiceQuestionContainer