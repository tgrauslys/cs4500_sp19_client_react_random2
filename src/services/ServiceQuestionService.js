export default class ServiceQuestionService {
    static instance = null;

    static getInstance() {
        if (ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService()
        }
        return this.instance
    }

    findServiceQuestionById = id =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/questions/${id}`)
            .then(response => response.json());
    findAllServiceQuestions = () =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/questions`)
            .then(response => response.json());
    findQuestionPage = (page, elements) =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/questions/paged?page=${page}&count=${elements}`)
            .then(response => response.json());
    linkQuestionAndAnswer = (questionId, answerId) =>
        fetch(`${process.env.REACT_APP_MIDDLE_TIER_URL}/api/questions/${questionId}/answers/${answerId}`, {
            method: "PUT"
        })
            .then(response => response.json());
}