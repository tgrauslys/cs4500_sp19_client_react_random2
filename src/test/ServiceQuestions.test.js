import React from 'react'
import ServiceQuestionContainer from '../containers/ServiceQuestionContainer'
import ServiceQuestionService from '../services/ServiceQuestionService'
import { BrowserRouter } from "react-router-dom";
import TestRenderer from 'react-test-renderer'

const serviceQuestionService = ServiceQuestionService.getInstance()
test('Render all service questions correctly', () => {
    const testRenderer = TestRenderer.create(
        <BrowserRouter>
            <ServiceQuestionContainer 
                service={serviceQuestionService}
                currentPage={0}
                itemCount={10}
                optionValues={[1, 2, 5, 10, 25, 50]}/>
        </BrowserRouter>
    )
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    let testInstance = testRenderer.root
    let nextPageBtn = testInstance.findByProps({className: "btn btn-primary next-page-btn"})

    nextPageBtn.props.onClick({
        target: {value: 0}
    })
    
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})