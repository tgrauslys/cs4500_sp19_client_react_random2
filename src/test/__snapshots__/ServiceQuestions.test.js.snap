// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render all service questions correctly 1`] = `
<div>
  <div>
    <h3>
      Service Questions
    </h3>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <td>
            <select
              onChange={[Function]}
              value={0}
            />
            
            
            <button
              className="btn btn-primary"
              disabled={true}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              1
            </button>
            <button
              className="btn btn-primary next-page-btn"
              onClick={[Function]}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              2
            </button>
            <button
              className="btn btn-secondary next-btn"
              onClick={[Function]}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              &gt;
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Render all service questions correctly 2`] = `
<div>
  <div>
    <h3>
      Service Questions
    </h3>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <td>
            <select
              onChange={[Function]}
              value={0}
            />
            
            
            <button
              className="btn btn-primary"
              disabled={true}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              1
            </button>
            <button
              className="btn btn-primary next-page-btn"
              onClick={[Function]}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              2
            </button>
            <button
              className="btn btn-secondary next-btn"
              onClick={[Function]}
              style={
                Object {
                  "margin": "2px",
                }
              }
              type="button"
            >
              &gt;
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
